app/controllers/api/v2/hosts_controller.rb

```diff
@@ -203,6 +203,21 @@ def facts
         render_message(e.to_s, :status => :unprocessable_entity)
       end
 
+      api :POST, "/hosts/:id/rebuild_config", N_("Rebuild TFTP, DNS DHCP configuration")
+      param :id, :identifier_dottable, :required => true
+      def rebuild_config
+        result = @host.rebuild_config
+        failures = []
+        result.each_pair do |key, value|
+          failures << key.to_s.upcase unless value
+        end
+        if failures.empty?
+          render :json => {:message => _("Configuration successfully rebuilt.")}, :status => :ok
+        else
+          render :json => {:message => _("Configuration rebuild failed for: %s.") % (failures.join(", "))}, :status => :unprocessable_entity
```

Comment:
I'm a bit worried about the level of error reporting here, should we pass/raise full exceptions back to the client if they failed?